flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/RocksDBKeyedStateBackend.java

```diff
@@ -867,9 +867,9 @@ public int numKeyValueStateEntries() {
     }
 
     @Override
-    public boolean requiresLegacySynchronousTimerSnapshots(CheckpointType checkpointType) {
+    public boolean requiresLegacySynchronousTimerSnapshots(SnapshotType checkpointType) {
         return priorityQueueFactory instanceof HeapPriorityQueueSetFactory
-                && checkpointType == CheckpointType.CHECKPOINT;
+                && !checkpointType.isSavepoint();
```

Comment:
Before changes it was only `CheckpointType.CHECKPOINT`, after changes it is `CheckpointType.CHECKPOINT` and `CheckpointType.FULL_CHECKPOINT`. Is it ok?

Reply:
Yes, actually it was a bug before.